src/containers/LearningResourcePage/components/LearningResourceTaxonomy.jsx

```diff
@@ -9,133 +9,466 @@
+      const subType = selectedResource.subtypes.find(
+        subtype => subtype.id === options[1],
+      );
+      val.push({
+        id: subType.id,
+        name: subType.name,
+        parentId: selectedResource.id,
+      });
+    }
+
+    this.stageTaxonomyChanges({
+      resourceTypes: val,
+    });
+  }
 
+  async getSubjectTopics(subjectid) {
+    if (
+      this.state.structure.some(
+        subject => subject.id === subjectid && subject.topics,
+      )
+    ) {
+      return;
+    }
     try {
-      const [resourceTypes, filters, topics, relevances] = await Promise.all([
-        fetchResourceTypes(model.language),
-        fetchFilters(model.language),
-        fetchTopics(model.language),
-        fetchRelevances(model.language),
+      this.updateSubject(subjectid, { loading: true });
+      const allTopics = await fetchSubjectTopics(subjectid);
+      const groupedTopics = groupTopics(allTopics);
+      this.updateSubject(subjectid, { loading: false, topics: groupedTopics });
+    } catch (e) {
+      handleError(e);
+    }
+  }
+
+  setPrimaryConnection(id) {
+    const { topics } = this.state.taxonomyChanges;
+
+    this.stageTaxonomyChanges({
+      topics: topics.map(topic => ({
+        ...topic,
+        primary: topic.id === id,
+      })),
+    });
+  }
+
+  async getResourceId() {
+    const { articleId, language } = this.props;
+    let resourceId = '';
+    const resource = await queryResources(articleId, language);
+    if (resource.length > 0) resourceId = resource[0].id;
+    return resourceId;
+  }
+
+  async fetchTaxonomy() {
+    const { language } = this.props;
+    try {
+      let { resourceId } = this.state;
+      if (!resourceId) {
+        resourceId = await this.getResourceId();
+      }
+      if (resourceId) {
+        const {
+          resourceTypes,
+          filters,
+          parentTopics,
+        } = await fetchFullResource(resourceId, language);
+
+        const topics = await Promise.all(
+          // Need to fetch each topic seperate because path is not returned in parentTopics
+          parentTopics.map(async item => {
+            const topicArticle = await fetchTopicArticle(item.id, language);
+            return {
+              ...topicArticle,
+              primary: item.isPrimary,
+              connectionId: item.connectionId,
+            };
+          }),
+        );
+
+        this.setState({
+          resourceId,
+          resourceTaxonomy: {
+            resourceTypes,
+            filter: filters,
+            topics,
+            isLoadingResource: false,
+          },
+          taxonomyChanges: {
+            topics,
+            resourceTypes,
+            filter: filters,
+          },
+        });
+      } else {
+        this.setState(prevState => ({
+          resourceTaxonomy: {
+            ...prevState.resourceTaxonomy,
+            isLoadingResource: false,
+          },
+        }));
+      }
+    } catch (e) {
+      handleError(e);
```

Comment:
We should give the user some feedback if something fails. Maybe isLoadingResouces and isLoadingTaxonomy can be renamed to `status` which can have the following possible values `'initial' | 'loading' | 'success' | 'error'`. Each value should be handled in render.